jkube-kit/jkube-kit-wildfly-jar/src/test/java/org/eclipse/jkube/wildfly/jar/enricher/WildflyJARHealthCheckEnricherTest.java

```diff
@@ -78,71 +82,170 @@ private void setupExpectations(Map<String, Map<String, Object>> jkubeConfig) {
+        assertThat(probe.getHttpGet().getScheme()).isEqualTo("HTTPS");
+        assertThat(probe.getHttpGet().getPort().getIntVal().intValue()).isEqualTo(1234);
+        assertThat(probe.getHttpGet().getPath()).isEqualTo("/foo/ready");
+        assertThat(probe.getInitialDelaySeconds().intValue()).isEqualTo(77);
+        assertThat(probe.getPeriodSeconds().intValue()).isEqualTo(30);
+        assertThat(probe.getSuccessThreshold().intValue()).isEqualTo(27);
+        assertThat(probe.getFailureThreshold().intValue()).isEqualTo(17);
+
+        probe = enricher.getStartupProbe();
+        assertThat(probe).isNull();
+    }
+
+    @Test
+    public void testWithCustomConfigurationComingFromConf_withWildflyJarAfter25_0shouldAdd_startupProbe() {
+        wildFlyJarDependencyWithVersion("26.1.1.Final");
+        Map<String, Object> jarConfig = new HashMap<>();
+        jarConfig.put("cloud", null);
+        final Map<String, Map<String, Object>> config = createFakeConfig("{\"readinessPath\":\"/foo/ready\","
+                + "\"livenessPath\":\"/foo/live\","
+                + "\"startupPath\":\"/foo/started\","
+                + "\"port\":\"1234\","
+                + "\"scheme\":\"HTTPS\","
+                +"\"livenessInitialDelay\":\"99\","
+                +"\"livenessPeriodSeconds\":\"10\","
                 +"\"readinessInitialDelay\":\"77\","
+                +"\"readinessPeriodSeconds\":\"30\","
+                +"\"startupInitialDelay\":\"10\","
+                +"\"startupPeriodSeconds\":\"10\","
                 +"\"failureThreshold\":\"17\","
                 +"\"successThreshold\":\"27\""
                 + "}");
         setupExpectations(project, jarConfig, config);
 
         final WildflyJARHealthCheckEnricher enricher = new WildflyJARHealthCheckEnricher(context);
         Probe probe = enricher.getLivenessProbe();
-        assertNotNull(probe);
-        assertNull(probe.getHttpGet().getHost());
-        assertEquals("HTTPS", probe.getHttpGet().getScheme());
-        assertEquals(1234, probe.getHttpGet().getPort().getIntVal().intValue());
-        assertEquals("/foo/live", probe.getHttpGet().getPath());
-        assertEquals(99, probe.getInitialDelaySeconds().intValue());
-        assertEquals(27, probe.getSuccessThreshold().intValue());
-        assertEquals(17, probe.getFailureThreshold().intValue());
+        assertThat(probe).isNotNull();
+        assertThat(probe.getHttpGet().getHost()).isNull();
+        assertThat(probe.getHttpGet().getScheme()).isEqualTo("HTTPS");
+        assertThat(probe.getHttpGet().getPort().getIntVal().intValue()).isEqualTo(1234);
+        assertThat(probe.getHttpGet().getPath()).isEqualTo("/foo/live");
+        assertThat(probe.getInitialDelaySeconds().intValue()).isEqualTo(99);
+        assertThat(probe.getPeriodSeconds().intValue()).isEqualTo(10);
+        assertThat(probe.getSuccessThreshold().intValue()).isEqualTo(27);
+        assertThat(probe.getFailureThreshold().intValue()).isEqualTo(17);
```

Comment:
Try to use chained assertions using assertj when asserting specific object. For example, here you can do something like this:

```java
        assertThat(probe)
            .isNotNull()
            .hasFieldOrPropertyWithValue("httpGet.host", null)
            .hasFieldOrPropertyWithValue("httpGet.scheme", "HTTPS")
            .hasFieldOrPropertyWithValue("httpGet.port", new IntOrString(1234))
            .hasFieldOrPropertyWithValue("httpGet.path", "/foo/live")
            .hasFieldOrPropertyWithValue("initialDelaySeconds", 99)
            .hasFieldOrPropertyWithValue("periodSeconds",  10)
            .hasFieldOrPropertyWithValue("successThreshold", 27)
            .hasFieldOrPropertyWithValue("failureThreshold", 17);
```